ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -498,7 +502,13 @@ void addToIndex(IndexEntry entry, FileSpan fileSpan) throws StoreException {
     } else {
       validIndexSegments.lastEntry().getValue().addEntry(entry, fileSpan.getEndOffset());
     }
-    journal.addEntry(entry.getValue().getOffset(), entry.getKey(), entry.getCrc());
+    if (config.storeAutoCloseLastLogSegmentEnabled && isForRecovery) {
```

Comment:
what if the config was True and then change to False? Can we just look at `isForRecovery`?